### ISSUE TITLE
FoxHttpRequestBody does not generate UTF-8 body

### ISSUE THREAD

**Issue by heffebaycay** [Viascom/FoxHttp#5]
While integrating the hipchat-api library, I noticed an encoding issue and opened https://github.com/Viascom/hipchat-api/issues/30. However, digging deeper shows that the problem is caused by FoxHttp... so I'm opening an issue here as well.

Here's the problem: if the request contains a non-ASCII character in its body, the character will not be properly encoded.

The culprit is the following method:
`ch.viascom.groundwork.foxhttp.body.request.FoxHttpRequestBody#writeBody`

```
DataOutputStream wr = new DataOutputStream(outputStream);
wr.writeBytes(json);
```

And here's the javadoc of `java.io.DataOutputStream#writeBytes`:
> Writes out the string to the underlying output stream as a
> sequence of bytes. Each character in the string is written out, in
> sequence, **by discarding its high eight bits**. If no exception is
> thrown, the counter `written` is incremented by the
> length of `s`.

Since internally `writeBytes(String)` calls `charAt`, it does not support UTF-8, which is why the character is not properly encoded in the request.

**Comment by patrickboesch** [comment]
Hi @heffebaycay 

Thanks for your bug report. Your analysis on Viascom/hipchat-api#30 is very nice and will help a lot. I will check our code and fix this bug as soon as possible.

Cheers, Patrick

**Comment by patrickboesch** [comment]
This issue is fixed in Version 1.3.1
